### PR TITLE
Improve Vive controller detection example

### PR DESCRIPTION
**Pull request by mncharity**
> Check all the gamepads.  Detect controllers by number of buttons, in
> addition to the current, but fragile, id string.  The Vive controller
> has 4 buttons.  The Occulus touch looks like 6.  An upper limit of 7
> avoids traditional gamepads.

I'm running a custom webvr stack, and my Vive controllers weren't being found.

The id string match was failing, because they aren't OpenVR gamepads.  Thus the additional functional check.  I kept the string match only because I can't run and regression test OpenVR.

Only 3 gamepads were being checked, since gamepads[0] is spec'ed to be null.  So with a headset (the Vive HMD has a button), and a single traditional gamepad, only one controller could be seen.  Add two Vive base stations (for their pose, and mode button), and a controller's odds weren't good.

The button test may give a false positive on the Samsung Gear VR HMD.  It functionally looks a lot like a Vive controller.  But, baby steps.

Thank you for your work.

### CODE REVIEW

**Comment by mrdoob** [comment]
Actually, @toji suggested that it would be better to just look for `gamepad.pose`.

**Comment by mncharity** [comment]
tl;dr 

    if ( gamepad && gamepad.pose != undefined &&
         gamepad.buttons.length >= 3 && gamepad.axes.length >= 2) {

    for ( var i = 0, j = 0; i < gamepads.length; i ++ ) {

----

Just looking for gamepad.pose, does seem an improvement over testing gamepad.id for 'OpenVR Gamepad'.  Somewhat less fragile.

But the issue remains.

I did considered *adding* a pose test to the buttons test.  There seemed one possible concern.  Flickering of controller tracking is fine.  VRPose properties can be transiently null.  Flickering of controller identity isn't.  Hopefully no one's gamepad.pose property will itself go transiently null.  But I've not seen any spec at all for how gamepad.pose is to behave.  Since the Vive and Touch controllers seem to have a currently unusual buttons profile, adding a pose test didn't seem to add much.  So I didn't.

But either way, with a pose test or without, the buttons test addresses a different issue.

The Vive headset has pose.  And a side button.  Also a button-like proximity sensor.

If the Vive headset is not currently returned by browser getGamepads(), it will be.  My stack already does.  Along with other non-controller things with buttons and pose.

*Hmm... drawing with your head in the vive paint example could be fun...*

I assume from the suggestion, that getGamepads(), on both Chrome and Firefox, *currently* does not expose the Vive headset as a gamepad?  Or do they order the gamepads list to keep the headset after any controllers?  I'm on linux, and so haven't checked.

If they both order the headset after the controllers, that sounds like a convention.  Or an accident becoming a convention.  Perhaps not pretty, but the Gamepad spec doesn't say anything about order.  Then just looking for gamepad.pose could be fine, and this PR unhelpful.  And I'll follow the crowd.

If both browsers simply haven't gotten around to the headset yet, then looking only for gamepad.pose, is only a temporary patch.  And my stack is the sick canary.

So what's the Right Thing?  The examples require gamepads which provide pose, and several buttons with recognizable identity.  We can't do much about recognizable identity until gamepad.map matures.  But I suggest we check for pose and buttons.

Interesting.  Perhaps that clarifies?  Reframing the test from "recognize Vive controllers" to "what is needed for Vive examples?"  A few buttons, a couple of axes, and pose.

Perhaps something like the tl;dr above.

**Comment by Mugen87** [comment]
Since this class also supports Oculus and Microsoft controller now, the PR is no longer valid.
